packages/react-ui/src/TableOfContent/useTOCReducer.js

```diff
@@ -0,0 +1,32 @@
+import { useReducer, useCallback } from "react";
+
+export const rootReducer = (state, action) => {
+  switch (action.type) {
+    case "setTitles":
+      return { titles: [...action.payload.map((title) => ({ ...title }))] };
```

Comment:
use slice or concat to make an array copy... don't overuse destructuring

Reply:
I want to make sure that the title items inside the array are new ones too.
Would this be ok ?
`return { titles: action.payload.map((title) => ({ ...title })) };`
(map indeed already returns a new array)

Reply:
why not but i'm not sure it is usefull to create new copy of items